internal/bminventory/inventory.go

```diff
@@ -1660,9 +1660,9 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 				params.NewHostParams.HostID, params.ClusterID.String(), err.Error())
 			b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityError,
 				"Failed to register host: cluster cannot accept new hosts in its current state", time.Now())
-			return installer.NewRegisterHostForbidden().
+			return installer.NewRegisterHostConflict().
```

Comment:
You can use `common.NewApiError(http.StatusConflict, err)`

Reply:
Nice...